packages/x-date-pickers/src/StaticDatePicker/StaticDatePicker.tsx

```diff
@@ -315,9 +315,15 @@ StaticDatePicker.propTypes = {
    * Available views.
    */
   views: PropTypes.arrayOf(PropTypes.oneOf(['day', 'month', 'year']).isRequired),
+  /**
+   * Years are displayed in ascending (chronological) order by default
+   * If `desc`, years are displayed in descending order
+   * @default 'asc'
+   */
+  yearsOrder: PropTypes.oneOf(['asc', 'desc']),
   /**
    * Years rendered per row.
-   * @default 3
+   * @default 4 on desktop, 3 on mobile
```

Comment:
If we want to improve the JSDoc for Static variants (it is currently not exactly correct 🙈 ), then it could make sense to use:
```
`4` when `displayStaticWrapperAs === 'desktop'`, `3` otherwise.
```

Reply:
In a lot of cases these were inherited from the ExportedYearCalendarProps, but those were not actually used in so many places. Rather yearsPerRow was added directly to the different pickers in which case it's easier to set the defaults. I will clean these up and maybe set yearsOrder directly also or at least make sure the defaults are properly overridden.

Reply:
I updated StaticDatePicker.types.tsx and StaticDateTimePicker.types.tsx to include overrides for this more specific default.

Reply:
MobileDatePicker and MobileDateTimePicker also feature overrides to default 3, though they aren't actually needed since ExportedYearCalendarProps exported default is now 3, but perhaps this is more clear. .ExportedYearCalendarProps could also have the info about desktop/mobile in the default but I guess it's supposed to be a more simple component compared to the picker).